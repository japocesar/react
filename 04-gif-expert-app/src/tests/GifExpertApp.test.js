import React from 'react';
import { shallow }  from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Tests in <GifExpertApp />', () => {
    test('should display the component', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('should display a categories list', () => {
        const categories = ['Simpsons', 'Wolverine'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    
});