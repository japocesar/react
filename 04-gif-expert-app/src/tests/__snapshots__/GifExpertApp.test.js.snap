// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> should display a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Simpsons"
      key="Simpsons"
    />
    <GifGrid
      category="Wolverine"
      key="Wolverine"
    />
  </ol>
</Fragment>
`;

exports[`Tests in <GifExpertApp /> should display the component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
